rpc/jsonrpc/client/ws_client.go

```diff
@@ -83,7 +94,13 @@ type WSClient struct { // nolint: maligned
 // functions for a detailed description of how to configure ping period and
 // pong wait time. The endpoint argument must begin with a `/`.
 // An error is returned on invalid remote. The function panics when remote is nil.
+// It uses DefaultWSOptions.
 func NewWS(remoteAddr, endpoint string, options ...func(*WSClient)) (*WSClient, error) {
+	return NewWSWithOptions(remoteAddr, endpoint, DefaultWSOptions(), options...)
+}
+
+// NewWSWithOptions allows you to provide custom WSOptions.
+func NewWSWithOptions(remoteAddr, endpoint string, opts WSOptions, options ...func(*WSClient)) (*WSClient, error) {
```

Comment:
Do we still need this options argument at the end?

Ah I see we have this reconnect callback function. Would it be possible to add that in `WSOptions`. I just think it looks a bit funky having two "option" argument.